src/inmanta/data/__init__.py

```diff
@@ -1394,6 +1412,37 @@ def get_contains_filter(
         value = cls._get_value(value)
         return (filter_statement, [value])
 
+    @classmethod
+    def get_combined_filter(
+        cls, name: str, value: Dict[QueryType, object], index: int, col_name_prefix: Optional[str] = None
+    ) -> Tuple[str, List[object]]:
+        """
+        Returns a tuple of a PostgresQL statement and any query arguments to filter on values that are contained in a given
+        collection.
+        """
+        cls.validate_field_name(name)
+        filter_statement: str
+        values: List[object]
+        (filter_statement, values) = cls._combine_filter_statements(
+            (cls.get_filter_for_query_type(query_type, name, val, index + i, col_name_prefix))
```

Comment:
if I understand correctly, this assumes that every filter has exactly one value. I see no reason why this assumption would hold. Nothing of this nature is documented and many of the filters have no value, in the future we may add some with multiple values.

This method is an accident waiting to happen.

Reply:
The typing/naming/documentation (`# The value describes a combination of other query types`)  suggest that this method combines filters, with no specific constraint on what kind of filters. Within that definition, it is broken, because it can not e.g. combine an is_null filter safely/correctly).